dpki/src/keypair.rs

```diff
@@ -81,6 +85,14 @@ impl KeyPair for SigningKeyPair {
         // Done
         Ok(SigningKeyPair::new(pub_key_b32, priv_sec_buf))
     }
+
+    fn new_from_self(&mut self) -> HcResult<Self> {
+        let mut buf = SecBuf::with_secure(self.private.len());
+        let pub_key = self.public();
+        let lock = self.private().read_lock();
+        buf.write(0, &**lock)?;
```

Comment:
perhaps we should implement clone for secbuf? https://github.com/holochain/holochain-rust/compare/clone-secbuf

Reply:
whoah!  Nice I never saw a PR for that!!  Would much prefer...

Reply:
I just made a PR for this.  Somehow I had it in my mind that SecBuf's couldn't be cloned.  Dunno why...